### ISSUE TITLE
support oracle workload capture

### ISSUE THREAD

**Issue by louishust** [TPC-Council/HammerDB#650]
**Is your feature request related to a problem? Please describe.**
Oracle support capture the workload. https://docs.oracle.com/en/database/oracle/oracle-database/21/ratug/capturing-a-database-workload.html. 

**Describe the solution you'd like**
Use the Oracle supported feature to capture the workload and replay the workload Instead of the trace file using Oracle Event 10046.
May be the workload can be replayed on other database such as postgresql?

**Comment by sm-shaw** [comment]
Oracle database replay is part of the [Real Application Testing](https://www.oracle.com/manageability/enterprise-manager/technologies/real-application-testing.html) product and is in a proprietary format.  Saying that, there are very good details on understanding this format here [Dissecting RAT](http://www.juliandyke.com/Presentations/DissectingRAT.ppt) and a good guide to RAT (also from Julian Dyke) is available here [Is RAT Worth Catching?](http://www.juliandyke.com/Presentations/IsRATWorthCatching.ppt). 

As RAT is an additional-licensed product, this includes both capture and replay - and therefore it does not seem to be worthwhile (even if development time was available) for HammerDB to implement only the replay part based on the guides above. 

Instead, trace files are not an additional cost-option, making this a preferable open source approach. Trace file replay was one of the first HammerDB features and there is a lot of potential for enhancement, including replaying against another database such as PostgreSQL.  Note however that with replaying OLTP database workloads there needs to be a degree of synchronization to ensure that workloads with multiple users are replayed in exactly the same order, otherwise they will generate multiple failures and therefore at present a converted trace file only provides a template for writing your own workload. 

So in summary, there is scope for anyone who wishes to enhance the Oracle trace file replay to improve the conversion and for running against additional databases, however there are no plans for the core development team to do this at the present time. Unless we have misunderstood something, it does not seem worthwhile trying to replay the format from RAT as you would have to license this product to both capture and replay workloads.

**Comment by louishust** [comment]
Thanks alot! @sm-shaw
